orchestrator/src/worker/controller/event_worker.rs

```diff
@@ -287,27 +312,81 @@ impl EventWorker {
     pub async fn run(&self) -> EventSystemResult<()> {
         let mut tasks = JoinSet::new();
         let max_concurrent_tasks = self.queue_control.max_message_count;
-        info!("Starting worker with concurrency limit: {}", max_concurrent_tasks);
+        info!("Starting worker with thread pool size: {}", max_concurrent_tasks);
 
         loop {
-            if tasks.len() < max_concurrent_tasks {
-                // Get message - this now blocks until a message is available
-                match self.get_message().await {
-                    Ok(message) => match self.parse_message(&message) {
-                        Ok(parsed_message) => {
-                            let worker = self.clone();
-                            tasks.spawn(async move { worker.process_message(message, parsed_message).await });
+            // Check if shutdown was requested at the start of each loop iteration
+            if self.is_shutdown_requested() {
+                info!("Shutdown requested, stopping message processing");
+                break;
+            }
+
+            tokio::select! {
+                biased;
+
+                // Immediate cleanup when tasks complete
+                Some(result) = tasks.join_next(), if !tasks.is_empty() => {
+                    Self::handle_task_result(result);
+
+                    if tasks.len() >= max_concurrent_tasks - 1 {
```

Comment:
should this be like:
```suggestion
                    if tasks.len() >= max_concurrent_tasks  {
```

Reply:
Actually it is in correct parameter if we remove the -1 then we need to remove the = too make this Simplifier have done it